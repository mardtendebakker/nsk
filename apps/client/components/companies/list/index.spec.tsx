import React from 'react';
import {
  render, fireEvent, waitFor,
} from '@testing-library/react';

import List from '.';

const mockAxios = {
  call: jest.fn(() => Promise.resolve()),
  get performing() { return false; },
  get data() { return { data: [{ id: 1 }], count: 1 }; },
};

jest.mock('../../../hooks/useAxios', () => () => mockAxios);
jest.mock('../form', () => function TestC() {
  return <div />;
});

const mockRouter = {
  push: jest.fn(),
  replace: jest.fn(),
  asPath: '',
};

jest.mock('next/router', () => ({
  useRouter: () => mockRouter,
}));

describe('List', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it('matches snapshot', () => {
    const { asFragment } = render(<List />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('makes first api call on mount', async () => {
    await waitFor(() => render(<List />));
    expect(mockRouter.replace).toBeCalledWith('?rowsPerPage=10', undefined, { scroll: false });
    expect(mockAxios.call).toBeCalledWith({ params: { skip: 0, take: 10 } });
  });
  it('handles search change properly', async () => {
    const { getByPlaceholderText } = render(<List />);
    fireEvent.change(getByPlaceholderText('search'), { target: { value: 'searchText' } });

    await waitFor(() => expect(mockRouter.replace).toHaveBeenCalledWith('?rowsPerPage=10', undefined, { scroll: false }));
    await waitFor(() => expect(mockRouter.replace).toHaveBeenCalledWith('?rowsPerPage=10&search=searchText', undefined, { scroll: false }));
    await waitFor(() => expect(mockAxios.call).toHaveBeenCalledWith({ params: { skip: 0, take: 10 } }));
    await waitFor(() => expect(mockAxios.call).toHaveBeenCalledWith({ params: { search: 'searchText', skip: 0, take: 10 } }));
  });
  it('doesn\'t search while performing', async () => {
    const { getByPlaceholderText } = render(<List />);
    jest.spyOn(mockAxios, 'performing', 'get').mockReturnValue(true);

    fireEvent.change(getByPlaceholderText('search'), { target: { value: 'searchText' } });

    await waitFor(() => expect(mockRouter.replace).not.toHaveBeenCalledWith('?rowsPerPage=10&search=searchText', undefined, { scroll: false }));
    await waitFor(() => expect(mockAxios.call).not.toHaveBeenCalledWith({ params: { search: 'searchText', skip: 0, take: 10 } }));
  });
  it('handles edit properly', async () => {
    const { getByTestId } = render(<List />);
    expect(getByTestId('edit-icon-button')).toHaveAttribute('href', '/companies/1');
  });
});
