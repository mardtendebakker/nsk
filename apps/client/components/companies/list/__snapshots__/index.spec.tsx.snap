// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-wvtwy6-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-19kjaan"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded MuiAccordion-gutters css-1086bdv-MuiPaper-root-MuiAccordion-root"
      >
        <div
          aria-expanded="false"
          class="MuiButtonBase-root MuiAccordionSummary-root MuiAccordionSummary-gutters css-1aipiv8-MuiButtonBase-root-MuiAccordionSummary-root"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAccordionSummary-content MuiAccordionSummary-contentGutters css-eqpfi5-MuiAccordionSummary-content"
          >
            <div
              class="MuiBox-root css-1ypcipt"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14"
                />
              </svg>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-1vwo10w-MuiFormControl-root-MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                    id=":r0:"
                    name="search"
                    placeholder="search"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-ihdtdm"
                    >
                      <span
                        class="notranslate"
                      >
                        â€‹
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorError MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorError css-fhzz8r-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                reset
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-colorPrimary css-wkp3g3-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                filter
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-uym7v6-MuiSvgIcon-root"
                  data-testid="ChevronRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-bz4dnt-MuiCollapse-root"
          style="min-height: 0px;"
        >
          <div
            class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
          >
            <div
              class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
            >
              <div
                class="MuiAccordion-region"
                role="region"
              >
                <hr
                  class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
                />
                <div
                  class="MuiAccordionDetails-root css-15v22id-MuiAccordionDetails-root"
                >
                  <div
                    class="MuiBox-root css-bq5sp3"
                  >
                    <div
                      class="MuiBox-root css-k008qs"
                    >
                      <div
                        class="MuiBox-root css-70qvj9"
                      >
                        <span
                          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                        >
                          <input
                            class="PrivateSwitchBase-input css-1m9pwf3"
                            data-indeterminate="false"
                            type="checkbox"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="CheckBoxOutlineBlankIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                        >
                          isCustomer
                        </p>
                      </div>
                      <div
                        class="MuiBox-root css-70qvj9"
                      >
                        <span
                          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                        >
                          <input
                            class="PrivateSwitchBase-input css-1m9pwf3"
                            data-indeterminate="false"
                            type="checkbox"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="CheckBoxOutlineBlankIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                        >
                          isSupplier
                        </p>
                      </div>
                      <div
                        class="MuiBox-root css-70qvj9"
                      >
                        <span
                          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                        >
                          <input
                            class="PrivateSwitchBase-input css-1m9pwf3"
                            data-indeterminate="false"
                            type="checkbox"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                            data-testid="CheckBoxOutlineBlankIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
                        >
                          isPartner
                        </p>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-tu93bp"
    />
    <div
      class="MuiBox-root css-bdvr1s"
    >
      <table
        class="MuiTable-root css-rqglhn-MuiTable-root"
      >
        <thead
          class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              name
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              company_kvk_nr
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              company_rsin_nr
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              isPartner
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              isSupplier
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1howxi1-MuiTableCell-root"
              scope="col"
            >
              isCustomer
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-6qfsqn-MuiTableCell-root"
              scope="col"
            >
              actions
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-hover css-1xe9h37-MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              --
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              --
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              --
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              --
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              --
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
            >
              --
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
            >
              <span
                aria-label="inactiveModuleMessage"
                class=""
                data-mui-internal-clone-element="true"
              >
                <span
                  class=""
                  data-mui-internal-clone-element="true"
                >
                  <a
                    aria-disabled="true"
                    class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall css-1rb4sk0-MuiButtonBase-root-MuiIconButton-root"
                    data-testid="edit-icon-button"
                    href="/companies/1"
                    tabindex="-1"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lr4af1-MuiSvgIcon-root"
                      data-testid="EditIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75z"
                      />
                    </svg>
                  </a>
                </span>
              </span>
            </td>
          </tr>
        </tbody>
      </table>
      <div
        class="MuiTablePagination-root css-1solmlx-MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
        >
          <div
            class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
          />
          <p
            class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
            id=":r4:"
          >
            rowsPerPage
          </p>
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-16c50h-MuiInputBase-root-MuiTablePagination-select"
          >
            <div
              aria-controls=":r5:"
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-labelledby=":r4: :r3:"
              class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
              id=":r3:"
              role="combobox"
              tabindex="0"
            >
              10
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value="10"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
          >
            1â€“1 of 1
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to previous page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowLeftIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                />
              </svg>
            </button>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
              disabled=""
              tabindex="-1"
              title="Go to next page"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="KeyboardArrowRightIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
